### PR TITLE
for node.js without a browser context

### PR DESCRIPTION
**Pull request by fuzhenn**
Hi, mourner

At first, really THANKS for this excellent work.

I'm working on porting simpleheat to node and rendering the heatmap with [node-canvas](https://github.com/Automattic/node-canvas/).

As simpleheat is designed for browsers at the first place,  I have done some revisions to make it work on node-canvas.

Everything works fine except the method to create canvas is not in the best way as the codes show.

Cheers

### CODE REVIEW

File: simpleheat.js [hunk 1]
```diff
@@ -128,5 +128,15 @@ simpleheat.prototype = {
...
         }
+    },
+    
+    _createCanvas:function() {
+        if (typeof document !== 'undefined') {
+            return document.createElement('canvas');
+        } else {
+            //works for node-canvas, but others?
+            //any better implementations?
+            return new this._canvas.constructor(1, 1);
```
Comment (mourner) [inline simpleheat.js]: What do those `1, 1` mean here?

File: simpleheat.js [hunk 2]
```diff
@@ -5,7 +5,7 @@ if (typeof module !== 'undefined') module.exports = simpleheat;
 function simpleheat(canvas) {
     if (!(this instanceof simpleheat)) return new simpleheat(canvas);
 
-    this._canvas = canvas = typeof canvas === 'string' ? document.getElementById(canvas) : canvas;
+    this._canvas = canvas = ( typeof canvas === 'string' && typeof document !== 'undefined' ) ? document.getElementById(canvas) : canvas;
```
Comment (mourner) [inline simpleheat.js]: Is the document check necessary? If you pass a node-canvas instance under Node, it's not a string anyway...

File: simpleheat.js [hunk 3]
```diff
@@ -84,7 +84,7 @@ simpleheat.prototype = {
         canvas.height = 256;
 
         for (var i in grad) {
-            gradient.addColorStop(i, grad[i]);
+            gradient.addColorStop(parseFloat(i), grad[i]);
```
Comment (mourner) [inline simpleheat.js]: it's faster to do `addColorStop(+i, ...)`

**Comment by fuzhenn** [comment]
@mourner 
done some updates as you advised.
Please have a check.

File: simpleheat.js (hunk 1 above)
Comment (fuzhenn) [inline simpleheat.js]: in node-canvas, canvas instances are usually created by:

``` javascript
var canvas = new Canvas(width, height)
```

so 1, 1 is the initial width and height. I removed it as node-canvas also has a default constructor without any parameter.

**Comment by fuzhenn** [comment]
Tested manually in node and all kinds of browsers.

Maybe consider to write some test codes? I can have a try. :)

**Comment by mourner** [comment]
Looks great! I'll look into adding some tests/CI.

**Comment by fuzhenn** [comment]
hi @mourner ,

Could you publish a new version on npm to take this PR on board?
Thank you very much!
:)

**Pull request merged**